Update field mapping documentation for Security Analytics (#2422)
opensearch-project/documentation-website: _security-analytics/sec-analytics-config/detectors-config.md

* Use the **Log type**, **Rule severity**, and **Source** dropdown menus to filter the rules you want to select from. 
* Use the **Search** bar to search for specific rules.

    To quickly select one or more known rules and dismiss others, first deselect all rules by moving the **Rule name** toggle to the left, then search for your target rule names and select each individually by moving its toggle to the right.
    {: .tip }

1. In the **Detector schedule** section, set how often the detector will run. Specify a unit of time and a corresponding number to set the interval.
1. Select the **Next** button in the lower-right corner of the screen to continue. The Configure field mapping page appears.

> ## Step 2. Make field mappings

Reviewer: Maybe "Create field mappings"?

Reviewer 2: oh and remember our style change to use gerunds for all section headings? So could be 
"Creating field mappings" then

Reviewer 3: Agree with Fanit: we are standardizing gerunds are for the TOC/nav pane.

> The field mapping step matches field names from the rule with field names from the log index being used to provide data. Correctly mapping the log field names with rule field names allows the system to accurately convey event data from the log to the detector and then use the data for triggering alerts.

Reviewer 2: suggestion:

To trigger alerts, the log event data needs to be mapped to the detector. The field mapping step maps the log field names from the index that provides the data with the rule field names to set up detectors to trigger alerts

Reviewer: Agree with Alice. I would just change the second sentence to "Creating field mappings maps the log field names..."

The data source (log index), log type, and detection rules specified in the first step determine which fields are available for mapping. For example, when "Windows logs" is selected as the log type, this parameter along with the specific detection rules determine the list of rule field names available for the mapping. Similarly, the selected data source (log index) determines the list of log field names that are available for the mapping.

### Pending field mappings

Once you navigate to the Configure field mapping page, the system attempts to automatically map fields between the two sources. Those field names that are not automatically mapped appear in the **Pending field mapping** table. In this table you can manually map rule fields to log fields.
<br><img src="{{site.url}}{{site.baseurl}}/images/Security/pending-mappings.png" alt="Field mapping example for pending mappings" width="900">

> This process requires that the user or administrator have some familiarity with the field names in the log index and an understanding of the data contained in those fields. Rule fields are named to promote their standardization and normalize them in the industry, and the names themselves are typically self explanatory. If a user has an understanding of the log data contained in a particular log field, the mapping is typically a simple and straightforward process.

Reviewer 2: maybe change "the user" to "you" ?
"If you have an understanding . . . "

Reviewer: Agree with Alice. Also,
This process requires **you** to have some familiarity... 
Also, not sure what "promote their standardization and normalize them in the industry" means in this context. Can we say "Rule fields are named according to industry standards"?

{: .note }

> * The Rule field name column lists field names based on the selected log type and the rules that have been assigned to the detector.

Reviewer 4: To be 100% factually correct, these rule fields are coming from ALL rules of selected log type/category (rule category == log type)

Reviewer 4: That is perfect! Thank you!

> * The log field name column includes a dropdown menu for each of the rule fields. Each dropdown menu contains a list of field names extracted from the log index.

Reviewer: Our style guide says "dropdown list"
```suggestion
* The log field name column includes a dropdown list for each of the rule fields. Each dropdown list contains a list of field names extracted from the log index.
```

Author: Good

> * Use the dropdown arrow to open the list of log fields and select the log field name that corresponds to the rule field name directly to the left in the Rule field name column. You can enter text in the **Select a mapping field** box at the top of the menu to search for names in the log field list. 

Reviewer: To map a rule field name to a log field name, use the dropdown arrow to open the list of log fields and select the log field name from the list. To search for names in the log field list, enter text in the **Select a mapping field** box.

Author: Good

<br><img src="{{site.url}}{{site.baseurl}}/images/Security/log-field.png" alt="Field mapping example for pending mappings" width="600">

> * Once the log field name is selected and mapped to the rule field name, the icon in the Status colum to the right changes to a green check mark.

Reviewer: ```suggestion
* Once the log field name is selected and mapped to the rule field name, the icon in the Status column to the right changes to a green check mark.
```

Author: Thanks

* Make as many matches between field names as possible to complete an accurate mapping for rule and log fields. 

### Default field mappings

The **Default mapped fields** table contains mappings that the system made automatically after defining the detector. As shown in the image that follows, when the field names are similar to one another the system can successfully match the two.
<br><img src="{{site.url}}{{site.baseurl}}/images/Security/default-mappings.png" alt="Field mapping example for pending mappings" width="900">

> Nevertheless, it's a good idea to review the mappings and verify that they are matched as expected and correct. If you find a mapping that doesn't appear to be accurate, you can use the dropdown menu as described in the **Pending field mappings** section above to correct the field mapping.  

Reviewer: ```suggestion
Nevertheless, it's a good idea to review the mappings and verify that they are correct and matched as expected. If you find a mapping that doesn't appear to be accurate, you can use the dropdown list as described in the [Pending field mappings](#pending-field-mappings) section to correct the field mapping.  
```

Author: I like it.

> After completing the mappings, select the **Next** button in the lower-right corner of the screen. The Set up alerts page appears and displays settings for an alert trigger.

Reviewer: ```suggestion
After completing the mappings, select the **Next** button in the lower-right corner of the screen. The **Set up alerts** page appears and displays settings for an alert trigger.
```

Author: OK, but we need to have a team talk on what we bold face in docs.

Reviewer: I say bold all UI elements

* Use the **Log type**, **Rule severity**, and **Source** dropdown menus to filter the rules you want to select from. 
* Use the **Search** bar to search for specific rules.

    To quickly select one or more known rules and dismiss others, first deselect all rules by moving the **Rule name** toggle to the left, then search for your target rule names and select each individually by moving its toggle to the right.
    {: .tip }

1. In the **Detector schedule** section, set how often the detector will run. Specify a unit of time and a corresponding number to set the interval.
1. Select the **Next** button in the lower-right corner of the screen to continue. The Configure field mapping page appears.

## Step 2. Create field mappings

> The field mapping step matches field names from the rule with field names from the log index being used to provide data. Creating field mappings allows the system to accurately convey event data from the log to the detector and then use the data for triggering alerts.

Reviewer 5: ```suggestion
The field mapping step matches field names from the rule with field names from the log index being used to provide data. Creating field mappings allows the system to accurately convey event data from the log to the detector and then use the data to trigger alerts.
```

> The data source (log index), log type, and detection rules specified in the first step determine which fields are available for mapping. For example, when "Windows logs" is selected as the log type, this parameter along with the specific detection rules determine the list of rule field names available for the mapping. Similarly, the selected data source (log index) determines the list of log field names that are available for the mapping.

Reviewer 5: ```suggestion
The data source (log index), log type, and detection rules specified in the first step determine which fields are available for mapping. For example, when "Windows logs" is selected as the log type, this parameter, along with the specific detection rules, determines the list of rule field names available for the mapping. Similarly, the selected data source (log index) determines the list of log field names that are available for the mapping.
```

#### A note on field names

> The field mapping process requires that you are familiar with the field names in the log index and have an understanding of the data contained in those fields. If you have an understanding of the log fields in the index, the mapping is typically a simple and straightforward process.

Reviewer 5: ```suggestion
The field mapping process requires that you are familiar with the field names in the log index and have an understanding of the data contained in those fields. If you have an understanding of the log fields in the index, the mapping is typically a straightforward process.
```

> Security Analytics takes advantage of pre-packaged Sigma rules for security event detection. Therefore, the rule field names are derived from a Sigma rule field standard. To make them easier to identify, however, we create aliases for the Sigma rule fields based on the open source Elastic Common Schema (ECS) specification. These alias rule field names are the field names used in these steps. 

Reviewer 5: ```suggestion
Security Analytics takes advantage of prepackaged Sigma rules for security event detection. Therefore, the rule field names are derived from a Sigma rule field standard. To make them easier to identify, however, we create aliases for the Sigma rule fields based on the open-source Elastic Common Schema (ECS) specification. These alias rule field names are the field names used in these steps. 
```

> Although the ECS rule field names are largely self explanatory, you can find pre-defined mappings of the Sigma rule field names with ECS rule field names for all supported log types in the GitHub Security Analytics repository. First navigate to the [OSMappings](https://github.com/opensearch-project/security-analytics/tree/main/src/main/resources/OSMapping) folder, select the folder named for the log type, and open the `fieldmappings.yml` file. For example, to see the Sigma rule fields that correspond to ECS rule fields for the Windows log type, open the [fieldmappings.yml file](https://github.com/opensearch-project/security-analytics/blob/main/src/main/resources/OSMapping/windows/fieldmappings.yml) in the **windows** folder.

Reviewer 5: ```suggestion
Although the ECS rule field names are largely self-explanatory, you can find predefined mappings of the Sigma rule field names to ECS rule field names for all supported log types in the GitHub Security Analytics repository. Navigate to the [OSMappings](https://github.com/opensearch-project/security-analytics/tree/main/src/main/resources/OSMapping) folder, select the folder named for the log type, and open the `fieldmappings.yml` file. For example, to see the Sigma rule fields that correspond to ECS rule fields for the Windows log type, open the [fieldmappings.yml file](https://github.com/opensearch-project/security-analytics/blob/main/src/main/resources/OSMapping/windows/fieldmappings.yml) in the **windows** folder.
```

### Pending field mappings

Once you navigate to the Configure field mapping page, the system attempts to automatically map fields between the two sources. Those field names that are not automatically mapped appear in the **Pending field mapping** table. In this table you can manually map rule fields to log fields.
<br><img src="{{site.url}}{{site.baseurl}}/images/Security/pending-mappings.png" alt="Field mapping example for pending mappings" width="900">

> * The Rule field name column lists field names based on all of the pre-packaged rules associated with the selected log type.

Reviewer 5: ```suggestion
* The Rule field name column lists field names based on all of the prepackaged rules associated with the selected log type.
```

> * The Rule field name column lists field names based on all of the pre-packaged rules associated with the selected log type.

Reviewer 5: The bulleted list should ideally be introduced by a brief sentence ending in a colon.

Author: Added ...
"While mapping fields, consider the following:"

* The log field name column includes a dropdown list for each of the rule fields. Each dropdown list contains field names extracted from the log index.
* To map a rule field name to a log field name, use the dropdown arrow to open the list of log fields and select the log field name from the list. To search for names in the log field list, enter text in the **Select a mapping field** box.
<br><img src="{{site.url}}{{site.baseurl}}/images/Security/log-field.png" alt="Field mapping example for pending mappings" width="600">
* Once the log field name is selected and mapped to the rule field name, the icon in the Status column to the right changes to a green check mark.
* Make as many matches between field names as possible to complete an accurate mapping for rule and log fields. 

### Default field mappings

> The **Default mapped fields** table contains mappings that the system made automatically after defining the detector. As shown in the image that follows, when the field names are similar to one another the system can successfully match the two.

Reviewer 5: ```suggestion
The **Default mapped fields** table contains mappings that the system created automatically after defining the detector. As shown in the image that follows, when the field names are similar to one another, the system can successfully match the two.
```

<br><img src="{{site.url}}{{site.baseurl}}/images/Security/default-mappings.png" alt="Field mapping example for pending mappings" width="900">

> Nevertheless, it's a good idea to review the mappings and verify that they are correct and matched as expected. If you find a mapping that doesn't appear to be accurate, you can use the dropdown list as described in the [Pending field mappings](#pending-field-mappings) section above to correct the field mapping.  

Reviewer 5: ```suggestion
Nevertheless, it's a good idea to review the mappings and verify that they are correct and matched as expected. If you find a mapping that doesn't appear to be accurate, you can use the dropdown list as described in the [Pending field mappings](#pending-field-mappings) section to correct the field mapping.  
```

Reviewer 5: Can we use something simpler than "Nevertheless"? It may not be understood by Global English speakers.

Author: I added this ...
"Although these automatic matches are normally dependable, it's still a good idea to review the mappings ..."

## Step 2. Make field mappings

> The field mapping step matches field names for the rule with field names for the log index being used to provide data. Correctly mapping the log field names with rule field names allows the system to accurately convey event data from the log to the detector and then use the data for triggering alerts.

Reviewer 4: nit: field names **_from_** the rules with field names **_from_** the log
